napi.h

```diff
@@ -7,10 +7,25 @@
 #include <string>
 #include <vector>
 
-#ifdef _NOEXCEPT
-  #define NAPI_NOEXCEPT _NOEXCEPT
+// If C++ exceptions are not explicitly enabled or disabled, enable them
+// if exceptions were enabled in the compiler settings.
+#if !defined(NAPI_CPP_EXCEPTIONS) && !defined(NAPI_DISABLE_CPP_EXCEPTIONS)
+  #if defined(_CPPUNWIND) || defined (__EXCEPTIONS)
+    #define NAPI_CPP_EXCEPTIONS
+  #else
+    #error Exception support not detected. \
+      Define either NAPI_CPP_EXCEPTIONS or NAPI_DISABLE_CPP_EXCEPTIONS.
+  #endif
+#endif
+
+#ifdef NAPI_CPP_EXCEPTIONS
```

Comment:
Are you sure this code isn’t kept best as it is? It’s not like `noexcept` will be incorrect if `NAPI_CPP_EXCEPTIONS` isn’t set, right?